src/itest/kotlin/nl/info/zac/itest/ZaakKoppelenRestServiceTest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package nl.info.zac.itest
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.kotest.core.spec.Order
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import nl.info.zac.itest.client.ZacClient
+import nl.info.zac.itest.config.ItestConfiguration.HTTP_STATUS_BAD_REQUEST
+import nl.info.zac.itest.config.ItestConfiguration.TEST_SPEC_ORDER_AFTER_ZAAK_CREATED
+import nl.info.zac.itest.config.ItestConfiguration.zaakProductaanvraag1Uuid
+import java.util.UUID
+
+private const val ROWS_DEFAULT = 10
+private const val PAGE_DEFAULT = 0
+
+/**
+ * This test assumes a zaak has been created in a previously run test.
+ */
+@Order(TEST_SPEC_ORDER_AFTER_ZAAK_CREATED)
+@Suppress("LargeClass")
+class ZaakKoppelenRestServiceTest : BehaviorSpec({
+    val zacClient = ZacClient()
+    val logger = KotlinLogging.logger {}
+
+    Given("ZAC Docker container is running and zaakafhandelparameters have been created") {
+        When("the searching for linkable cases happy path") {
+            val response = zacClient.findLinkableCases(
+                zaakProductaanvraag1Uuid,
+                "HOOFDZAAK",
+                PAGE_DEFAULT,
+                ROWS_DEFAULT
+            )
+            Then("the response should be a 400 HTTP response") {
+                response.code shouldBe HTTP_STATUS_BAD_REQUEST
+                val responseBody = response.body!!.string()
+                logger.info { "Response: $responseBody" }
+            }
+        }
+
+        When("the searching for linkable cases with an invalid case UUID") {
```

Comment:
So far our convention is to mostly only test the happy path in our integration tests and not the unhappy paths with 4xx and 5xx responses. I'm not against testing this at all but if we start doing this more it will mean that our itests may become even more complex and harder to work with.

Reply:
Hmm, ok, works for me.